components/script/dom/vrframedata.rs

```diff
@@ -51,10 +51,14 @@ impl VRFrameData {
         };
 
         unsafe {
-            framedata.left_proj.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
-            framedata.left_view.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
-            framedata.right_proj.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
-            framedata.right_view.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
+            let _ = TypedArray::<Float32>::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                MutableHandleObject::from_marked_location(framedata.left_view.get_unsafe()));
+            let _ = TypedArray::<Float32>::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                MutableHandleObject::from_marked_location(framedata.left_proj.get_unsafe()));
+            let _ = TypedArray::<Float32>::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                MutableHandleObject::from_marked_location(framedata.right_proj.get_unsafe()));
+            let _ = TypedArray::<Float32>::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                MutableHandleObject::from_marked_location(framedata.right_view.get_unsafe()));
```

Comment:
We should add a `handle_mut()` method to `Heap` instead of using from_marked_location everywhere.